extmod/modnetwork.c

```diff
@@ -56,17 +56,50 @@ char mod_network_country_code[2] = "XX";
 #error "MICROPY_PY_NETWORK_HOSTNAME_DEFAULT must be set in mpconfigport.h or mpconfigboard.h"
 #endif
 
-char mod_network_hostname[MICROPY_PY_NETWORK_HOSTNAME_MAX_LEN] = MICROPY_PY_NETWORK_HOSTNAME_DEFAULT;
+MP_REGISTER_ROOT_POINTER(mp_obj_t mod_network_hostname_override);
```

Comment:
I don't think this needs to be a root pointer because it's just a qstr object.  So how about storing it just in a `static qstr mod_network_hostname_override;` variable?

Reply:
Yes, definitely, makes the intent much clearer too. Updated the branch.